### PR TITLE
feat(i18n): add loaded namespaces and isLoaded into useTranslation 

### PR DESCRIPTION
**Pull request by philibea**
Add a new state namespaces on Provider, this state re-present namespace that are already loaded.
Add isLoaded on useTranslation hook, this params will help to avoid multi-rerender when it's not needed.
Add ReactDom unstable_batchedUpdates  to minimize number of renders

### CODE REVIEW

File: packages/use-i18n/package.json [hunk 1]
```diff
@@ -30,15 +30,18 @@
     "intl-format-cache": "^4.3.1",
     "intl-messageformat": "^9.5.3",
     "intl-pluralrules": "^1.2.2",
-    "prop-types": "^15.7.2"
+    "prop-types": "^15.7.2",
+    "react-dom": "^17.0.2"
```
Comment (chambo-e) [inline packages/use-i18n/package.json:34]: Not needed :)

File: packages/use-i18n/src/index.js [hunk 2]
```diff
@@ -235,15 +250,15 @@ const I18nContextProvider = ({
...
+      formatNumber,
       loadTranslations,
-      locales,
       namespaceTranslation,
+      namespaces,
       relativeTime,
       relativeTimeStrict,
-      setLocales,
       setTranslations,
+      supportedLocales,
```
Comment (chambo-e) [inline packages/use-i18n/src/index.js:261]: What the difference between supportedLocales and locales ?

File: packages/use-i18n/src/index.js [hunk 3]
```diff
@@ -114,15 +130,22 @@ const I18nContextProvider = ({
...
       const preparedValues = prefix ? prefixKeys(`${prefix}.`)(values) : values
 
-      setTranslations(prevState => ({
-        ...prevState,
-        ...{
-          [currentLocale]: {
-            ...prevState[currentLocale],
-            ...preparedValues,
+      // avoid a lot of render when async update
+      ReactDOM.unstable_batchedUpdates(() => {
```
Comment (vincentaudebert) [inline packages/use-i18n/src/index.js:134]: any idea why is it "unstable"?

File: packages/use-i18n/src/index.js (hunk 2 above)
Comment (philibea) [reply packages/use-i18n/src/index.js:261]: Before we use a state and it's not used anymore

File: packages/use-i18n/src/index.js (hunk 3 above)
Comment (vincentaudebert) [reply packages/use-i18n/src/index.js:134]: Ok got it in your article

Although this function is supposedly “unstable”, React apparently intends to address this in the following versions.

    “In future versions (probably React 17 and later), React will batch all updates by default so you won’t have to think about this”, according to Dan Abramov.

**Review by vincentaudebert — APPROVED** [review]

File: packages/use-i18n/src/index.js (hunk 3 above)
Comment (philibea) [reply packages/use-i18n/src/index.js:134]: It's unstable for the moment but it's use in React, Redux library from a long time. 
In a future upgrade of react, i thinks it's will not necessary

**Review by chambo-e — APPROVED** [review]

**Pull request merged**